### PR TITLE
Fix AssociationMap in TrackRefitter

### PR DESCRIPTION
**Pull request by aysent**
Starting with CMSSW 75X initialization of AssociationMap needs to be modified in some cases.

I followed comments in DataFormats/Common/interface/AssociationMap.h and discussion in:

https://hypernews.cern.ch/HyperNews/CMS/get/edmFramework/3540.html

to make it work for TrackRefitter package.

### CODE REVIEW

**Comment by slava77** [comment]
@aysent please make a PR for 81X.
merging in 80X may have lower priority for things like this.

@rovere @VinInn  @aysent 
if TracksToTrajectories is regularly used, it better be added to one of the AlCa or similar steps in relval matrices. It wouldn't take 8 months to discover the interface change in that case.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by VinInn** [comment]
AssociationMaps are a relict of the past that introduces huge performance penalties in particular in recent releases. By definition Track and Trajectory are in one by one correspondence and produced by the same producers.
just loop over them from 0 to .size()

**Comment by slava77** [comment]
On 2/23/16 10:15 PM, Vincenzo Innocente wrote:
> AssociationMaps are a relict of the past that introduces huge
> performance penalties in particular in recent releases. By definition
> Track and Trajectory are in one by one correspondence and produced by
> the same producers.
> just loop over them from 0 to .size()
> 

Hi Vincenzo,

TrackingTools/TrackRefitter is in TRK purview, if I'm not mistaken
May I suggest that after operational functionality is restored in this PR,
someone follows up with a more appropriate implementation.

Maybe a JIRA ticket will help this happen.
Please let me know.


		--slava

File: TrackingTools/TrackRefitter/plugins/TracksToTrajectories.cc [hunk 1]
```diff
@@ -126,14 +126,14 @@ void TracksToTrajectories::produce(Event& event, const EventSetup& setup){
...
-  
-  // Association map between Trajectory and Track
-  auto_ptr<TrajTrackAssociationCollection> trajTrackMap(new TrajTrackAssociationCollection);
  
   // Get the RecTrack collection from the event
   Handle<reco::TrackCollection> tracks;
   event.getByToken(theTracksToken, tracks);
   
+  // Association map between Trajectory and Track
+  auto_ptr<TrajTrackAssociationCollection> trajTrackMap(new TrajTrackAssociationCollection(trajectoryCollectionRefProd, tracks));
```
Comment (cvuosalo) [inline TrackingTools/TrackRefitter/plugins/TracksToTrajectories.cc]: auto_ptr is deprecated. Please use unique_ptr or shared_ptr instead.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
@cmsbuild please test
Comparisons got stuck in Queued state

**Comment by cvuosalo** [comment]
+1

For #13445 8441d5301602b6e9338cc233af0faa42823febb6

Small fix for track-trajectory map in track refitter. There should be no change in monitored quantities.
#13446 is the 81X version of this PR.

The code change is satisfactory, and Jenkins tests against baseline CMSSW_8_0_X_2016-02-23-2300 show no significant differences, as expected.

**Pull request merged**